cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryLogDecider.java

```diff
@@ -64,6 +64,11 @@ private boolean executionTimeForLogging(long executionTimeInMilliseconds) {
 
     private boolean typeForLogging(Query query) {
         logger.trace("Checking if query type is for logging.");
+        boolean logAll = slowQueryConfiguration.queryTypesToLog().contains("ALL");
+        boolean queryTypeMatches = slowQueryConfiguration.queryTypesToLog().contains(query.statementType().toString());
+        if (!logAll && !queryTypeMatches) {
+            return false;
+        }
         if (query.statementType() == StatementType.SELECT || query.statementType() == StatementType.UPDATE) {
             return true;
         }
```

Comment:
You should check here if type is for logging as well. 

`return logAll || queryTypeMatches;`

And than you do not need two ifs.